### PR TITLE
Fix segfault in table function file while inserting into it

### PR DESCRIPTION
**Pull request by stavrolia**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Fixed issue #8108
Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Fixed segfault in table function file while inserting into file that doesn't exist. Now in this case file would be created and then insert would be processed.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Test needed.

**Comment by stavrolia** [comment]
> Test needed.


I forget to make this PR a draft or stay [WIP] in name when decide to add test in another commit. Sorry that I wasted your time on obvious comment.

**Comment by stavrolia** [comment]
Failed tests don't influenced by this PR

**Review by KochetovNicolai — APPROVED** [review]
LGTM

File: dbms/src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -150,7 +148,12 @@ StorageFile::StorageFile(
                 poco_path = Poco::Path(db_dir_path_abs, poco_path);
 
             const std::string path = poco_path.absolute().toString();
-            paths = listFilesWithRegexpMatching("/", path);
+            if (path.find_first_of("*?{") == std::string::npos)
```
Comment (tavplubix) [inline dbms/src/Storages/StorageFile.cpp]: @stavrolia, looks like server still crashes with
`insert into table function file('non}{exist.csv', 'CSV', 'val1 UInt32') values (1)`

**Pull request merged**